edb/ir/staeval.py

```diff
@@ -176,7 +176,13 @@ def evaluate_OperatorCall(
 
         args.append(arg_val)
 
-    value = eval_func(*args)
+    try:
+        value = eval_func(*args)
+    except TypeError:
+        raise UnsupportedExpressionError(
```

Comment:
I think the rule should be that `statypes` *must* implement ordering (all types in EdgeQL are orderable), and then this should be an ISE.

Reply:
We can definitely support ordering for everything, but I don't think it is going to necessarily be the case that every operator that we support for some type in staeval will necessarily be supported for all of them.

Reply:
I've been reluctant to add more operators to staeval given the subtle differences in results between Python and Postgres, especially when it comes to math.  

`TypeError` is not very specific so the approach in this patch risks masking implementation bugs unrelated to actual support or non-support.

Reply:
(I did just implement support though)